src/effects/audiounits/AudioUnitEffect.cpp

```diff
@@ -1152,26 +1151,21 @@ bool AudioUnitEffect::ProcessFinalize()
 {
    mOutputList.reset();
    mInputList.reset();
-
    return true;
 }
 
 size_t AudioUnitEffect::ProcessBlock(EffectSettings &,
    const float *const *inBlock, float *const *outBlock, size_t blockLen)
 {
-   for (size_t i = 0; i < mAudioIns; i++)
-   {
-      mInputList[0].mBuffers[i].mNumberChannels = 1;
-      mInputList[0].mBuffers[i].mData = const_cast<float*>(inBlock[i]);
-      mInputList[0].mBuffers[i].mDataByteSize = sizeof(float) * blockLen;
-   }
+   assert(PackedArrayCount(mInputList) >= mAudioIns);
+   for (size_t i = 0; i < mAudioIns; ++i)
+      mInputList[i] = { 1, static_cast<UInt32>(sizeof(float) * blockLen),
+         const_cast<float*>(inBlock[i]) };
```

Comment:
I might be wrong but... this change is not equivalent? the previous code loops on the mBuffers[i] of mInputList[0]; the new code loops on mInputLists instead, always assigning only the first mBuffer.

Reply:
Besides, I would keep the original assigning by field, instead of the aggregate assignment. Easier to read and more resilient to future changes in the struct.

Reply:
I hope you are reading my commit comments as well as code comments.  I explained in one:

"Note that AudioUnitGetProperty can retrieve less of the data than are available
    from the plug-in if it doesn't first check with AudioUnitGetPropertyInfo and
    it passes a small size.
    
    The SDK appears to do this to allow versioning of the structures while not
    breaking code written to prior versions."

For that reason, I suspect that what you suggest is an unnecessary caution:  Apple would be smart enough not to do
that, breaking Audacity and any other audio program getting reinstalled on upgraded versions of the operating system.

But then I shouldn't read their minds either.  So I will insert another commit in this branch that adds some more
C++ wrappers around the SDK structures, in the header `AudioUnitUtils.h`, and I will amend some other commits to
use those wrappers rather than directly using the SDK.  Wait for it...

Reply:
> I assume that the change in what is being indexed here, is the correction of the formal error mentioned in the commit message.

The old version was subscripting `AudioBufferList::mBuffers` (`CoreAudioBaseTypes.h`) after it had allocated it "wrong" though harmlessly wrong (a bit of overallocation)

```
struct AudioBufferList
{
    UInt32      mNumberBuffers;
    AudioBuffer mBuffers[1]; // this is a variable length array of mNumberBuffers elements
};
```

The "proper" thing to do is not to allocate an array of `AudioBufferList` structures, but to do the things that `PackedArrayAllocateCount` does:  make space for one `AudioBufferList ` plus sufficient for extra `AudioBuffer` objects, whose number is known only after first calling `AudioUnitGetPropertyInfo`.

Reply:
So maybe I'm just being pedantic fixing what never caused a bug, but, it seemed to be a bit of old confusion of thought that is now better expressed.